openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -771,6 +774,64 @@ def create_container_version(
     return container_version
 
 
+class ChildrenEntitiesAction(Enum):
+    """Possible actions for children entities"""
+
+    APPEND = "append"
+    REMOVE = "remove"
+    REPLACE = "replace"
+
+
+def get_next_entity_list(
+    learning_package_id: int,
+    last_version: ContainerVersion,
+    publishable_entities_pks: list[int],
+    entity_version_pks: list[int | None] | None,
+    entities_action: ChildrenEntitiesAction = ChildrenEntitiesAction.REPLACE,
+) -> EntityList:
+    """
+    Creates next entity list based on the given entities_action.
+
+    Args:
+        learning_package_id: Learning package ID
+        last_version: Last version of container.
+        publishable_entities_pks: The IDs of the members current members of the container.
+        entity_version_pks: The IDs of the versions to pin to, if pinning is desired.
+        entities_action: APPEND, REMOVE or REPLACE given entities from/to the container
+
+    Returns:
+        The newly created entity list.
+    """
+    if entity_version_pks is None:
+        entity_version_pks: list[int | None] = [None] * len(publishable_entities_pks)  # type: ignore[no-redef]
+    if entities_action == ChildrenEntitiesAction.APPEND:
+        # get previous entity list rows
+        last_entities = last_version.entity_list.entitylistrow_set.only("entity_id", "entity_version_id")
```

Comment:
I forgot something: the existing `entity_list` needs to be ordered by `order_num` to maintain the original sort order, both here and with REMOVE below.

Reply:
Nice catch! Updated here: ddbae25683bd and also set default ordering in EntityListRow model to avoid this issue in future: 0c7730a9eb9a